### DISCUSSION TITLE
Test games for legacy player

### DISCUSSION THREAD

**Discussion by corinnewo** [ji-devs/ji-cloud#1780] [General]
The following games have been built to systematically test all the options available for each activity. The options chosen are in text on the screens. @dakom 

Houdini: https://jitap.net/activities/geno/play/
**Sound board**: https://jitap.net/activities/genr/play/ 
**Say something**:  https://jitap.net/activities/gen8/play/
**Video**: https://jitap.net/activities/genx/play/
**Ask a question**: https://jitap.net/activities/geoi/play/testing-ask-a-question-legacy-player
**Puzzle**: https://jitap.net/activities/gepc/play/test-puzzles-for-legacy-player (On web, it seems that none of the settings actually have an effect)
**Talk or type**: https://jitap.net/activities/gepa/play/test-talk-or-type-for-legacy-player 
(There are some issues with the web player on Ji Tap that I noticed for Talk or Type: 

1. If the box was set to talk on iPad, then on web it is automatically type.
2. On ipad - tapping houdini cuts off sound, and toggles hide or show.
    On web - clicking houdini has not effect whatsoever - does not play sound, does not cut off main sound, does not hide/show
3. Hints on or off - 
      on iPad, hints on will show a random extra letter when there is a mistake; hints off will simply color the box red and you can try again. 
      on web, hints on or off, will behave the same way - as if there are hints on!)
Screen capture of talk or type on iPad: https://jewishinteractive.slack.com/files/U04TP8STM/F02M41GSZRQ/rpreplay_final1636707272.mp4

Soundboard settings:
![Settings soundboard](https://user-images.githubusercontent.com/16290361/139395976-42300a39-431e-4a84-9a87-11749cf561ee.PNG)

Video settings:
![Video settings](https://user-images.githubusercontent.com/16290361/139396520-2d4d1e8a-97cd-4d89-aaec-86ee883ef35b.PNG)

**Comment by dakom** [comment]
local dev url after `cargo make local-main module/legacy/play` is `http://localhost:4104/module/legacy/play/debug/debug?game_id=GAME_ID&slide_index=SLIDE_INDEX`

comparative Ji Tap player is `https://jitap.net/media/webplayer/webplayer.html?structureJson=https://jitap.net/store/api/album/GAME_ID/structure`

On the jigzi side, the following mirrors the above. 

- Puzzle: 
  - jig: https://sandbox.jigzi.org/jig/play/58e559b6-5d16-11ec-b989-0b8848fe9f42?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17822&slide_index=0

- Talk n Type: 
  - jig: https://sandbox.jigzi.org/jig/play/58c21582-5d16-11ec-aaff-9b1a2ba39e6e?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17820&slide_index=0
  
- Ask a Question: 
  - jig: https://sandbox.jigzi.org/jig/play/59060b48-5d16-11ec-b989-8306f94a2876?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17792&slide_index=0

- Soundboard:
  - jig: https://sandbox.jigzi.org/jig/play/58ee9d8c-5d16-11ec-bbef-a797e94ef8ab?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17765&slide_index=0
  
- Video: 
  - jig: https://sandbox.jigzi.org/jig/play/590c4918-5d16-11ec-aaff-c730cf73cab5?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17771&slide_index=0

- Say Something: 
  - jig: https://sandbox.jigzi.org/jig/play/58a4ef3e-5d16-11ec-92e1-b7da831d40e6?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17746&slide_index=0
    
- Corinne Houdini: 
  - jig: https://sandbox.jigzi.org/jig/play/58d835a6-5d16-11ec-aa84-a7c792e1b5a9?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17762&slide_index=0
  
- David Houdini: 
  - jig: https://sandbox.jigzi.org/jig/play/58f9c7de-5d16-11ec-aaff-d7b134486f14?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=17736&slide_index=0
  
- Let's Read about Tet: 
  - jig: https://sandbox.jigzi.org/jig/play/b9cc44d0-5e9f-11ec-aced-3b709d1f3584?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=7556&slide_index=0

- משפחה:  החיות שלנו: 
  - jig: https://sandbox.jigzi.org/jig/play/b9d46412-5e9f-11ec-aced-93651d382cd5?direction=ltr&display_score=false&track_assessments=false&drag_assist=false&is_student=false&draft=false
  - module: https://sandbox.jigzi.org/module/legacy/play/debug/debug?game_id=16248&slide_index=0

Please keep in mind the "quirks" listed here: https://github.com/ji-devs/ji-cloud/discussions/1787

And that we are generating the above while working across modules... it's not impossible that we'll need to re-generate as code changes. Of course, before final QA we'll re-generate them all, or better yet push them as proper jigs ;)

**Reply by corinnewo** [reply]
@chayaml Did you manage to run the Jigzi games?

**Reply by chayaml** [reply]
Yes, the Jigs all mirrored the JI Tap games, aside from a few minor things:

- Soundboard: 
1.  The tooltip is hidden by the bottom bar
2. Clicking on present Jumps to next screen where not instructed

- Corrine Houdini:  
1. the Bet doesn't have a sound
2. By Veshamru "read to me" Screen doesn't match properly - small screen

- Ask a Question : Houdini doesnt work (letter Dalet)

- Video:
1. the background doesn't %100 fit screen when enlarging it.
2. Pressing pause button on bottom right should pause video?

**Reply by dakom** [reply]
Thanks for checking! Going to address them from my perspective... ~~@corinnewo - I think there is only one full-on bug to file..~~ **edit: fixed!**

**Soundboard**
> The tooltip is hidden by the bottom bar

If you mean the jig player controls, that is essentially by design (i.e. not a bug for me to fix). Only solution I can think of would be to have a dedicated controlbar like what we see on most video players. In any case, this is a general Jigzi UI issue, not specific to legacy player

> Clicking on present Jumps to next screen where not instructed

I did not experience that... can you please zoom with @corinnewo and confirm?

Note that the jump must happen when an audio finishes, so if another sounds interrupts, the jump will be cancelled

**Corrine Houdini:**

> the Bet doesn't have a sound

Oh no... opening an issue for me to check into that: https://github.com/ji-devs/ji-cloud/issues/2013 (**edit: fixed!**)

> By Veshamru "read to me" Screen doesn't match properly - small screen

Yeah... audio and animation are inherently through different pipelines and will be out of sync. This is due to TinyTap's design and keeping with the technical underpinnings to render as gif. Their web player has the same problem, though with caching and luck it may or may not appear more in sync.

A future improvement would be to render as transparent video and line up the tracks perfectly, but that has complications when trying to target all browsers and platforms.

I'll add some of this explanation to the [quirks doc](https://github.com/ji-devs/ji-cloud/discussions/1787) too.

**Ask a Question : Houdini doesnt work (letter Dalet)**

Although this differs from Tiny Tap, this is by design, @corinnewo said the question audio should take precedence

(**edit: see below.. it's not entirely by design, but, it's a discussion point**)

**Video:**

> the background doesn't %100 fit screen when enlarging it.

We are limited by what youtube allows us to do

> Pressing pause button on bottom right should pause video?

Nice idea! It's tricky to implement though because the bottom controls are always there while the youtube controls are unpredictable... it's doable but tricky... anyway, I think it's outside the scope of legacy player specifically and would be a general jigzi player thing.

**Reply by dakom** [reply]
oh... wait, for ask a question it's a race condition and I tried to set it up so that the houdini _should_ win... making it definitively win here would be pretty ugly. we'd basically have to delay it from playing and have an interrupt to stop that delay, it's going to be messy.

But anyway the whole mechanism is nonsensical since clicking on it means both things should happen. I think the experience is essentially on par with Tiny Tap in spirit if not in exact action... in other words, let it be whatever it is, even if it's sometimes arbitrary. If we really need to revisit houdini + hotspot in ask a question, we can always come back to it. @corinnewo - your call

**Reply by dakom** [reply]
Ahh.. I see what happened... there was an update since those links were created :)

I am going to re-publish these jigs and change the links. that should fix bet...

**Reply by dakom** [reply]
Waiting on a fix for publish... may have to pick it up again tomm

**Reply by dakom** [reply]
fixed :)

**Comment by corinnewo** [comment]
Notes on what we must take into account with the player:

1. Background music
2. Mode - if quiz or not and the settings for a quiz:
    **Time limit** for either ACTIVITY or GAME in minutes and seconds
    **Wrong answer limit** (number starting at 1)
3. Shuffle if on, i think it shuffles the screens randomly. Cannot do this if there are jump to screens in the game.

**Reply by dakom** [reply]
Yes, in other words, we need a brief on how to map Ji Tap Games to Jigs. If the feature doesn't exist on Jigs, it can't be implemented.

**Reply by corinnewo** [reply]
We have everything in Jigzi except for the Shuffle. Background music, time limit, etc are all settings at either JIG or activity level.

**Reply by dakom** [reply]
Great. So I guess there needs to be a decision of either adding Shuffle, or leaving it out completely

**Reply by dakom** [reply]
I don't understand what 2. means here... can you please clarify? Or we could jump on zoom sometime...

**Reply by corinnewo** [reply]
Definitely can zoom on this. I have not yet analysed the quiz well, but I think that it very similar to the quiz mode in Jigzi.

**Comment by corinnewo** [comment]
@dakom When we bring across the games, can we redirect from existing game links? For example, Shalom Learning references their Ji Tap games in documentation. Can we preserve the original link and redirect to JIgzi games?

**Reply by dakom** [reply]
It's probably doable
